### PR TITLE
Implemented code for Wind/Pollution Rose dialogue

### LINKED ISSUE #5965: New Wind/Pollution Rose dialogue
This is a new dialogue to implement the Wind/Pollution Rose plot from the OPenair package.  It is to go into the new Circular menu.
It is described in Chapter 5 of the Openair [Manual](https://davidcarslaw.com/files/openairmanual.pdf) together with examples of its use.
I start the design below, but there are many options which I hope that @HawardKetoyoMsatsi can explore to add, in particular to the sub-dialogue structure.  It can be merged in stages as facilities are implemented.

So it starts with the main dialogue having the same structure as the existing Wind Rose.  It is called `Wind/Pollution Rose`.  The main selector is the same as are the 2 single controls on the right, namely `Wind Speed: `and `Wind Direction:`.  
Under that is a checkbox called Compare.  Default is unchecked.  If checked then there are 2 more single receivers called `Wind Speed 2: `and `Wind Direction 2:`. 

Then (as in the existing wind rose plot) there is Facet (Optional).  Under that is Facet 2 (Optional):
The existing Wind Rose has the Titles, etc on the main dialogue.  Move this here to the Windrose Options dialogue here.
There is a `Windrose Options` button.
Of course the Save Graph control and the bottom Comment/OK control.
A checkbox under the options button `Paddle`.  Default is checked
A checkbox under that `Normalise`. Default is unchecked.
A drop-down with label `Summary`.  Options are Proportion (default), Mean, Count.

There is more to be put on the sub-dialogue.  But this could be merged (and tested) with these.

I hope @HawardKetoyoMsatsi can check the design, and consider the sub-dialogue, while @Wycklife

### PR DESCRIPTION
**Pull request by HawardKetoyoMsatsi**
Fixes #5965
@africanmathsinitiative/developers @rdstern this is ready for review.
this replaces PR #6126 which had build errors.

### CODE REVIEW

**Comment by rdstern** [comment]
@haward This is looking very good.  I have suggestions for mainly just cosmetic changes:
a) Move Pollutant up (above the 2nd set of wd and ws.  And say Pollutant (Optional):
b) If just ws and wd are completed then enable Ok and run windrose rather than pollution rose.
c) Have a checkbox below the Pollutant saying "Compare two sets"  Default is unchecked and then the ws2 and wd2 are not visible.  If checked, then they are as you have it now.

**Review by rdstern — CHANGES_REQUESTED** [review]
@HawardKetoyoMsatsi this was probably still waiting for more changes from you, but I looked anyway.
a) Good change you made in the order of the controls and enabling it to function as an ordinary wind-rose plot.
b) Still good to add (Optional) after Pollutant.
c) Change the tab order, so it goes to pollutant after wind direction.  Then to the facets.
d) Consider a checkbox - as mentioned before for the 2nd ws/wd - unless that is too complicated

**Comment by HawardKetoyoMsatsi** [comment]
@rdstern I was still working on it when you reviewed it but am now done, please take a look once more.

**Review by rdstern — APPROVED** [review]
@HawardKetoyoMsatsi great - looks good now.  @lloyddewit over to you!

**Comment by lloyddewit** [comment]
@shadrackkibet please could you review? thanks

**Review by shadrackkibet — CHANGES_REQUESTED** [review]

**Comment by HawardKetoyoMsatsi** [comment]
@shadrackkibet Please have a look at the corrections done.

**Comment by lloyddewit** [comment]
@shadrackkibet Please could you check the changes from @HawardKetoyoMsatsi above, thanks

**Comment by shadrackkibet** [comment]
Yes, I will, thanks for the reminder @lloyddewit. @HawardKetoyoMsatsi sorry for keeping you on this. I should have checked a long time ago!  Please send reminders when things stay for long.

File: instat/dlgWindPollutionRose.vb [hunk 1]
```diff
@@ -1,3 +1,151 @@
...
+' along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-End Class
\ No newline at end of file
+Imports instat.Translations
+
+Public Class dlgWindPollutionRose
+    Private bFirstLoad As Boolean = True
+    Private bReset As Boolean = True
+    Private bRcodeSet As Boolean = False
```
Comment (shadrackkibet) [inline instat/dlgWindPollutionRose.vb:22]: Remove because it's not used anywhere?

File: instat/dlgWindPollutionRose.vb [hunk 2]
```diff
@@ -1,3 +1,151 @@
...
+        ucrChkCompareTwoSets.AddParameterPresentCondition(True, "wd2")
+
+        ucrChkPaddle.SetText("Paddle")
+        ucrChkPaddle.SetParameter(New RParameter("paddle", 7))
+        ucrChkPaddle.SetValuesCheckedAndUnchecked("TRUE", "FALSE")
+
+        ucrInputStatistic.SetParameter(New RParameter("statistic", 8))
+        dctStatistic.Add("prop.count", Chr(34) & "prop.count" & Chr(34))
+        dctStatistic.Add("prop.mean", Chr(34) & "prop.mean" & Chr(34))
+        dctStatistic.Add("abs.count", Chr(34) & "abs.count" & Chr(34))
```
Comment (shadrackkibet) [inline instat/dlgWindPollutionRose.vb:84]: Will it be better to use long names here? e.g Proportion count, Proportion mean, absolute mean?

File: instat/dlgWindPollutionRose.vb [hunk 3]
```diff
@@ -1,3 +1,151 @@
...
+
+        ucrReceiverFacetTwo.SetParameter(New RParameter("type2", 11))
+        ucrReceiverFacetTwo.Selector = ucrSelectorWindPollutionRose
+        ucrReceiverFacetTwo.SetParameterIsString()
+
+        ucrReceiverWindDirection2.SetLinkedDisplayControl(lblWindDirection2)
+        ucrReceiverWindSpeed2.SetLinkedDisplayControl(lblWindSpeed2)
+        ucrChkCompareTwoSets.AddToLinkedControls({ucrReceiverWindDirection2, ucrReceiverWindSpeed2}, {True}, bNewLinkedAddRemoveParameter:=True, bNewLinkedHideIfParameterMissing:=True)
+
+        ucrSaveGraph.SetPrefix("wind_pollution_rose")
```
Comment (shadrackkibet) [inline instat/dlgWindPollutionRose.vb:108]: It's dangerous to use the same prefix as the function name. The function will be overridden when you save the plot. I suspect the dialog will be unusable after that.

**Review by shadrackkibet — CHANGES_REQUESTED** [review]
@HawardKetoyoMsatsi check some additional points here. If you could fix then I will approve. Thanks.

**Comment by shadrackkibet** [comment]
@HawardKetoyoMsatsi 8 days ago I reviewed this. Could you give an update on the state of this PR?

**Comment by HawardKetoyoMsatsi** [comment]
Hi, @shadrackkibet I have made the proposed changes, kindly take a look.

File: instat/dlgWindPollutionRose.vb (hunk 2 above)
Comment (HawardKetoyoMsatsi) [reply instat/dlgWindPollutionRose.vb]: Yes, you are right, the labels will look better that way.

File: instat/dlgWindPollutionRose.vb [hunk 4]
```diff
@@ -1,3 +1,150 @@
...
+        If bFirstLoad Then
+            InitiliseDialog()
+            bFirstLoad = False
+        End If
+
+        If bReset Then
+            SetDefaults()
+        End If
+        SetRcodeForControls(bReset)
+        bReset = False
```
Comment (shadrackkibet) [inline instat/dlgWindPollutionRose.vb:35]: I notice the translation class is missing here!

**Review by shadrackkibet — CHANGES_REQUESTED** [review]

File: instat/dlgWindPollutionRose.vb (hunk 4 above)
Comment (HawardKetoyoMsatsi) [reply instat/dlgWindPollutionRose.vb:35]: Done please take a look

**Review by shadrackkibet — APPROVED** [review]
@rdstern @lloyddewit over to you.

File: instat/static/InstatObject/R/stand_alone_functions.R [hunk 5]
```diff
@@ -2315,6 +2315,28 @@ other_rose_plots <- function(data, type1_col_name, type2_col_name, date_col_name
...
+  type <-  "default"
+  if (!missing(type1_col_name) && !missing(type2_col_name)) {
+    type <- c(type1_col_name, type2_col_name)
+  }
+  if (missing(type1_col_name) && !missing(type2_col_name)) {
+    type <- type2_col_name
+  }
+  if (!missing(type1_col_name) && missing(type2_col_name)) {
+    type <- type1_col_name
+  }
```
Comment (lloyddewit) [inline instat/static/InstatObject/R/stand_alone_functions.R:2329]: Only one of these if statements will ever be true. Would it be more readable to use `Else If` for the 2nd and 3rd if statements. No changes, needed, I'll approve anyway, it's jiust a thought.

**Review by lloyddewit — APPROVED** [review]

**Comment by lloyddewit** [comment]
@rdstern If you can test/approve again, then we can merge, thanks

**Review by rdstern — APPROVED** [review]

**Pull request merged**